docs/specification/new-spec/consensus.md

```diff
@@ -0,0 +1,210 @@
+the gossiping protocol, processes also send auxiliary messages that inform peers about the 
+executed steps of the core consensus algorithm (`NewRoundStepMessage` and `CommitStepMessage`), and 
+also messages that inform peers what votes the process has seen (`HasVoteMessage`,
+`VoteSetMaj23Message` and `VoteSetBitsMessage`).
+ 
+We now describe the content of each message exchanged during Tendermint consensus protocol.
+
+## ProposalMessage
+ProposalMessage is sent when a new block is proposed. It is a suggestion of what the 
+next block in the blockchain should be.
+```
+type ProposalMessage struct {
+	Proposal *types.Proposal
+}
+```
+### Proposal
+Proposal contains height and round for which this proposal is made, BlockID as a unique identifier of proposed
+block, timestamp, and two fields (POLRound and POLBlockID) that are needed for termination of the consensus.
+The message is signed by the validator private key.  
+
+```
+type Proposal struct {
+	Height           int64            
+	Round            int              
+	Timestamp        time.Time        
+	BlockID          BlockID      
+	POLRound         int              
+	POLBlockID       BlockID          
+	Signature        crypto.Signature 
+}
+```
+
+NOTE: In the current version of the Tendermint, the consensus value in proposal is represented with
+PartSetHeader, and with BlockID in vote message. It should be aligned as suggested in this spec as 
+BlockID contains PartSetHeader.   
+
+## VoteMessage
+VoteMessage is sent to vote for some block (or to inform others that a process does not vote in the current round).
+
+```
+type VoteMessage struct {
+	Vote *types.Vote
+}
+```
+
+### Vote
+Vote contains validator's information (validator address and index), height and round for which the vote is sent,
+vote type, blockID if process vote for some block (`nil` otherwise) and a timestamp when the vote is sent. 
+The message is signed by the validator private key.
+
+```
+type Vote struct {
+	ValidatorAddress data.Bytes       
+	ValidatorIndex   int              
+	Height           int64            
+	Round            int              
+	Timestamp        time.Time        
+	Type             byte             
+	BlockID          BlockID          
+	Signature        crypto.Signature 
+}
+```
+## BlockPartMessage
+BlockPartMessage is sent when gossipping a piece of the proposed block. It contains height, round 
+and the block part. 
+
+```
+type BlockPartMessage struct {
+	Height int64
+	Round  int
+	Part   *types.Part
+}
+```
+
+## ProposalHeartbeatMessage 
+ProposalHeartbeatMessage is sent to signal that a node is alive and waiting for transactions 
+to be able to create a next block proposal.
+
+```
+type ProposalHeartbeatMessage struct {
+    Heartbeat *types.Heartbeat
+}
+```
+
+### Heartbeat
+Heartbeat contains validator information (address and index),
+height, round and sequence number (?). It is signed by the private key of the validator. 
```

Comment:
(?)

Reply:
I was wondering what is the sequence field, but it's probably as it says sequence number :)